views/appdeskpicker/main.view.php

```diff
@@ -44,7 +45,18 @@
         var loadAppdesk = function ($wijtab, target) {
             var $tab = $(target.tab);
             if (!$tab.data('loaded')) {
-                $(target.panel).load($tab.data('appdesk-url'));
+                $(target.panel).load($tab.data('appdesk-url'), function (responseData) {
+                    try {
+                        var data = JSON.parse(responseData);
+                        if (data.hasOwnProperty('error')) {
+                            // An error happened, but it's glitchy and the error structure is inconsistent
+                            // across all appdesks, so we have to output a generic error message...
+                            $(target.panel).html(<?= json_encode(__('You are not allowed to access to this application.')) ?>);
```

Comment:
As I said in the PR, I don't really have any other choice since the appdesk doesn't output a standard error structure, so I don't have any other solution. And the error key outputted by some appdesk is crap and very confusing for the user.

Reply:
Ok.

And what about a more generic error ? like `We are unable to load this application. Sorry about that. Please try again or contact a website administrator.`.